src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +53,176 @@ class MissingProjectMembership extends React.Component<Props, State> {
       },
       {
         success: () => {
           this.setState({
             loading: false,
             error: false,
           });
+          addSuccessMessage(t('Request to join team sent.'));
         },
         error: () => {
           this.setState({
             loading: false,
             error: true,
           });
-          addErrorMessage(t('There was an error while trying to leave the team.'));
+          addErrorMessage(t('There was an error while trying to request access.'));
         },
       }
     );
   }
 
-  renderJoinTeam(team: Team, features: Set<string>) {
+  renderJoinTeam(team: string, features: Set<string>) {
+    const teamStatus = TeamStore.getBySlug(team);
+
     if (!team) {
       return null;
     }
     if (this.state.loading) {
-      return <a className="btn btn-default btn-loading btn-disabled">...</a>;
-    } else if (team.isPending) {
-      return <a className="btn btn-default btn-disabled">{t('Request Pending')}</a>;
+      if (features.has('open-membership')) {
+        return <StyledButton busy>Join Team</StyledButton>;
+      }
+      return <StyledButton busy>Request Access</StyledButton>;
+    } else if (teamStatus?.isPending) {
+      return <StyledButton disabled>Request Pending</StyledButton>;
     } else if (features.has('open-membership')) {
       return (
-        <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
+        <StyledButton
+          priority="primary"
+          type="button"
+          onClick={this.joinTeam.bind(this, team)}
+        >
           {t('Join Team')}
-        </a>
+        </StyledButton>
       );
     }
     return (
-      <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
+      <StyledButton
+        priority="primary"
+        type="button"
+        onClick={this.joinTeam.bind(this, team)}
+      >
         {t('Request Access')}
-      </a>
+      </StyledButton>
     );
   }
 
-  renderExplanation(features: Set<string>) {
-    if (features.has('open-membership')) {
-      return t('To view this data you must one of the following teams.');
-    } else {
-      return t(
-        'To view this data you must first request access to one of the following teams:'
-      );
-    }
-  }
-
-  renderJoinTeams(features: Set<string>) {
+  getTeamsForAccess() {
+    const request: string[] = [];
+    const pending: string[] = [];
     const teams = this.state.project?.teams ?? [];
+    const team = teams.map(tm => TeamStore.getBySlug(tm.slug));
+    team.map(tm => (tm?.isPending ? pending.push(tm!.slug) : request.push(tm!.slug)));
+    return [request, pending];
+  }
 
-    if (!teams.length) {
-      return (
-        <EmptyMessage>
-          {t(
-            'No teams have access to this project yet. Ask an admin to add your team to this project.'
-          )}
-        </EmptyMessage>
-      );
-    }
+  handleChange = (evt: string) => {
+    const input = evt;
+    this.setState({team: input});
+  };
```

Comment:
I would name this more specific like `handleChangeTeam`

It looks like `SelectField.onChange` callback's first argument is the `value` of the selected option and not a form event like native form elements. In this case, I would change the parameter name to be `team` so that we can simplify the `setState` as well.